qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -160,15 +165,27 @@ def __init__(
         self._eval_count = 0
 
     @property
-    def initial_point(self) -> Sequence[float] | None:
+    def initial_point(self) -> Optional[Union[np.ndarray, List[np.ndarray]]]:
         """Returns initial point."""
         return self._initial_point
 
     @initial_point.setter
-    def initial_point(self, initial_point: Sequence[float]):
+    def initial_point(self, initial_point: Optional[Union[np.ndarray, List[np.ndarray]]]):
         """Sets initial point"""
         self._initial_point = initial_point
 
+    @property
+    def optimizer(self) -> Union[Optimizer, List[Optimizer]] | Union[Minimizer, List[Minimizer]]:
+        """Returns optimizer"""
+        return self._optimizer
+
+    @optimizer.setter
+    def optimizer(
+        self, optimizer: Union[Optimizer, List[Optimizer]] | Union[Minimizer, List[Minimizer]]
+    ):
+        """Sets optimizer"""
+        self._optimizer = optimizer
+
```

Comment:
We're moving to use public class variables where possible (so just `self.optimizer`) without any private variables (`self._optimizer`). If there's no extra logic involved, I think we don't need these setters and leave it as before.